web/tests/functional/report_viewer_api/test_run_data.py

```diff
@@ -171,3 +174,67 @@ def test_sort_run_data(self):
         # contains special characters.
         sort_mode = RunSortMode(RunSortType.NAME, Order.ASC)
         self._cc_client.getRunData(None, None, 0, sort_mode)
+
+    def test_analysis_info(self):
+        """
+        Test that storing runs to the server records the executed analyzer
+        command and the list of checkers present and executed.
+        """
+        workspace = os.environ["TEST_WORKSPACE"]
+        runs = self.__get_runs("test_files*%")
+        self.assertEqual(len(runs), 1,
+                         "There should be one run for this test.")
+        run = runs[0]
+        run_id = run.runId
+
+        analysis_infos = self._cc_client.getAnalysisInfo(
+            AnalysisInfoFilter(run_id, None, None), 1, 0)
+        self.assertEqual(len(analysis_infos), 1,
+                         "An analysis_info must be recorded for the run!")
+
+        info = analysis_infos[0]
+        cmd = info.analyzerCommand
+        print(run_id, analysis_infos)
+
+        self.assertTrue(workspace in cmd,
+                        "The name of the test workspace should be part of "
+                        "the report directory, found in the cmdline.")
+        # Ensure that the tests here are up-to-date with what's in __init__.py.
+        self.assertTrue("-d core.StackAddressEscape" in cmd,
+                        "A disabled checker is needed for this test to work!")
+        self.assertTrue("-d unix.Malloc" in cmd,
+                        "A disabled checker is needed for this test to work!")
+        self.assertTrue("-d clang-diagnostic" in cmd,
+                        "A disabled checker is needed for this test to work!")
+        self.assertTrue("-e clang-diagnostic-division-by-zero" in cmd,
+                        "An enabled checker is needed for this test to work!")
+
+        checkers = info.checkers
+
+        def assertChecker(analyzer, checker):
+            self.assertTrue(checkers[analyzer][checker].enabled)
+
+        def assertNotChecker(analyzer, checker):
+            self.assertFalse(checkers[analyzer][checker].enabled)
+
+        assertNotChecker("clangsa", "alpha.cplusplus.MismatchedIterator")
```

Comment:
What should I do in this case? Do we have a well-known earliest version that is supposed to be supported? Otherwise, #4116. It's incredibly stupid that running these tests requires you to have an analysis environment...

Reply:
The `MismatchedIteratorChecker` has been part of LLVM since version **`8.0`**. This was the earliest commit I could identify where it was introduced, and has not been moved between packages ever since: https://github.com/llvm/llvm-project/commit/21583b733af14997718ab5aaf2a926821d479dd7#diff-e1b5fd68572f20de546330e9ed6537c1276f09f78f7ba7aac45b4fd887b7cc7bR1491

Reply:
I removed tests for `alpha.`, and instead opted for checking that a `debug.` checker is indeed disabled, as it almost certainly always is. 🙂